include/nudb/impl/basic_store.ipp

```diff
@@ -13,6 +13,7 @@
 #include <boost/assert.hpp>
 #include <cmath>
 #include <memory>
+#include <boost/filesystem.hpp>
```

Comment:
This is a hefty dependency to add, and we are using so little of it. I would prefer if we had our own functions to perform the needed operations, e.g. `path_cat`:
```
// Append an HTTP rel-path to a local filesystem path.
// The returned path is normalized for the platform.
std::string
path_cat(
    beast::string_view base,
    beast::string_view path)
{
    if(base.empty())
        return std::string(path);
    std::string result(base);
#ifdef BOOST_MSVC
    char constexpr path_separator = '\\';
    if(result.back() == path_separator)
        result.resize(result.size() - 1);
    result.append(path.data(), path.size());
    for(auto& c : result)
        if(c == '/')
            c = path_separator;
#else
    char constexpr path_separator = '/';
    if(result.back() == path_separator)
        result.resize(result.size() - 1);
    result.append(path.data(), path.size());
#endif
    return result;
}
```

Reply:
We can maybe fall back to `boost::filesystem` if we can't identify the platform